src/main/java/com/amazon/corretto/hotpatch/HotPatchMain.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.corretto.hotpatch;import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.jvmstat.monitor.MonitoredHost;
+import sun.jvmstat.monitor.MonitoredVm;
+import sun.jvmstat.monitor.MonitoredVmUtil;
+import sun.jvmstat.monitor.VmIdentifier;
+
+import static com.amazon.corretto.hotpatch.Constants.LOG4J_FIXER_AGENT_VERSION;
+import static com.amazon.corretto.hotpatch.Logger.log;
+
+public class HotPatchMain {
+    public static void main(String[] args) throws Exception {
+        Logger.setVerbose(args);
+
+        String pid[];
+        if (args.length == 0) {
+            MonitoredHost host = MonitoredHost.getMonitoredHost((String)null);
+            Set<Integer> pids = host.activeVms();
+            pid = new String[pids.size()];
+            int count = 0;
+            for (Integer p : pids) {
+                MonitoredVm jvm = host.getMonitoredVm(new VmIdentifier(p.toString()));
+                String mainClass = MonitoredVmUtil.mainClass(jvm, true);
+                if (!myName.equals(mainClass)) {
+                    log(p + ": " + mainClass);
+                    pid[count++] = p.toString();
+                }
+            }
+            if (false && count > 0) {
+                log("Patch all JVMs? (y/N) : ");
+                BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+                String answer = in.readLine();
+                if (!"y".equals(answer)) {
+                    System.exit(1);
+                    return;
+                }
```

Comment:
Remove this block

Reply:
Done, I wanted to remove that piece of dead code before, but I didn't want to make it confusing on this PR.
What I have done in that function is add a check to validate the pids are proper numbers. I will port the logic for args parsing that I had on my first iteration of this change (not this PR, the previous iteration) later